contents/docs/libraries/hono.mdx

```diff
@@ -0,0 +1,96 @@
+
+```js file=index.ts
+import { env } from 'hono/adapter'
+import { createMiddleware } from 'hono/factory'
+import { PostHog } from 'posthog-node'
+
+const posthogServerMiddleware = createMiddleware(async (c, next) => {
+  const { POSTHOG_PUBLIC_KEY } = env<{ POSTHOG_PUBLIC_KEY:string }>(c)
+  const posthog = new PostHog(POSTHOG_PUBLIC_KEY, { host: 'https://us.i.posthog.com' })
+
+  posthog.capture({
+      distinctId: 'distinct_id_of_user', // Their user id or email
+      event: 'user_did_something',
+  })
+  await posthog.shutdown() // TIP: On program exit, call shutdown to stop pending pollers and flush any remaining events
+  await next()
+})
+```
+
+<CalloutBox icon="IconInfo" title="Shutdown">
+
+Make sure to always call `posthog.shutdown()` after capturing events from the server-side. PostHog queues events into larger batches, and this call forces all batched events to be flushed immediately.
+
+</CalloutBox>
+
+Next, apply this middleware to your app. This middleware will be called on every request handled by Hono, denoted by the wildcard `*`.
+
+```ts file=index.ts
+// use the middleware on all (*) routes.
+app.use('*', posthogServerMiddleware)
+```
+
+In your routes, you can pass information to be captured by your middleware using [context `set()` and `get()` pair](https://hono.dev/docs/api/context#set-get)
+
+<CalloutBox icon="IconInfo" title="Cloudflare Pages">
+
+Cloudflare Pages uses its own middleware system that is different from Hono's middleware. Follow the [Hono documentation on Cloudflare Pages](https://hono.dev/docs/getting-started/cloudflare-pages#cloudflare-pages-middleware) to handle middleware. 
+
+</CalloutBox>
+
+## Error tracking
+
+Hono uses [`app.onError`](https://hono.dev/docs/api/exception#handling-httpexception) for root-level error handling. You can take advantage of this for [Error tracking](/docs/error-tracking).
+
+```ts file=index.ts
+import { env } from 'hono/adapter'
+import { PostHog } from 'posthog-node'
+
+app.onError((err, c) => {
+  const { POSTHOG_PUBLIC_KEY } = env<{ POSTHOG_PUBLIC_KEY:string }>(c)
+  const posthog = new PostHog(POSTHOG_PUBLIC_KEY, { host: 'https://us.i.posthog.com' })
+  
+  posthog.captureException(new Error(err.message, { cause: err }), 'user_distinct_id_with_err_rethrow', {
+    path: c.req.path,
+    method: c.req.method,
+    url: c.req.url,
+    headers: c.req.header(),
+    // ... other properties
+  })
+  posthog.shutdown() // TIP: On program exit, call shutdown to stop pending pollers and flush any remaining events
+  // other error handling logic
+  return c.text('Internal Server Error', 500)
+})
+```
+
+## Next steps
+
+For any technical questions for how to integrate specific PostHog features into Hono, have a look at our [Node](/docs/libraries/node) SDK docs for concepts such as:
+- [Capture custom events, identify users, and more.](/docs/libraries/node#capturing-events)
+- [Feature flags including variants and payloads.](/docs/libraries/node#feature-flags)
```

Comment:
Identifying isn't that important in this context, identify broadly is more of a frontend concept
```suggestion
- [Capturing custom events, setting properties, and more.](/docs/libraries/node#capturing-events)
- [Setting up feature flags including variants and payloads.](/docs/libraries/node#feature-flags)
```